### ISSUE TITLE
Named value refs should format float values better

### ISSUE THREAD

**Issue by agrrr3** [freeorion/freeorion#3247]
Currently, value refs defined as NamedReal are displayed like `12.000000`

Example: ![mouseover](https://user-images.githubusercontent.com/11386908/100291247-8abd2080-2f7d-11eb-807e-ab91e979c674.png)

Depending on circumstances it should be `12`, or `12.00` or `12,00` or `1200%`

That formatting decision should be made in the stringtables file where it is used.
So it would be great if we had something like a formatting syntax to apply to the values.

Or the poor mans version: make it display `12.00`

**Comment by Oberlus** [comment]
I'd like to set a maximum number of decimals (say 2), and then

When there are no decimals (it can be represented by an integer), remove all decimal zeros:
12.000000 -> 12

When there are decimals, remove all decimal zeros after last non-zero decimal and truncate to maximum number of decimals if needed:
12.5 -> 12.5
12.333333 -> 12.33
12.000400 -> 12.00

With roundings?
12.666667 -> 12.67

**Comment by agrrr3** [comment]
@Oberlus yes, the format suggestions look good for a poor mans implementation. if we make it a bit more fancy by allowing the client to set the number of decimals I guess we should put that into the UI rules section (or make it a functional stringtable entry?)

What would be the canonical way to do that formatting in cpp?
